osidb/migrations/0210_affect_v1_materialized_view.py

```diff
@@ -0,0 +1,185 @@
+# Generated by Django 4.2.25 on 2025-11-04 11:03
+
+from django.db import migrations
+
+
+CREATE_AFFECT_V1_MATERIALIZED_VIEW = """
+DROP VIEW IF EXISTS affect_v1;
+
+CREATE MATERIALIZED VIEW affect_v1 AS
```

Comment:
I have been far away from osidb for some time - just a db observation that every materialized view incurs a cost at some point - I am sure the pros/cons have been balanced to use them here but I aways question why (same feeling for triggers, partioning, etc). Mostly asking for satisfying curiosity then seeing anything wrong with the approach.

Reply:
the cost is having to recompute the view periodically to not have data that is too old, but even recomputing it every 5 minutes beats the current view which has to be computed each time v1 is queried. not saying it is the best solution but it is a quick win that will at least make it workable while further optimizations are done.

Reply:
The goal is to keep investigating and improving the view and eventually, once we're happy with its performance, turn it back into a dynamic view.

However this performance issue has OSIDB and downstream tooling dead in the water, so this is a stop-gap solution that works fairly well and requires little effort.

Also note that the API which uses the view is deprecated and is intended to be thrown out, so even improving the view might be wasted effort.

Reply:
thx for the context - MV can also cause their own performance issues - assuming that has been factored in - all makes sense. +1 for 'keep investigating'.

FWIW if it helps I can take a look at stage (with that query) to see if there are any indexes that could be set with this kind of query ( assuming **pg_stat** enabled on stage db). Or with anything else if it helps.